### ISSUE TITLE
Failing test: [firefox] › page/page-click.spec.ts:162:3 › should click offscreen buttons

### ISSUE THREAD

**Issue by dgozman** [microsoft/playwright#20993]
https://github.com/microsoft/playwright/actions/runs/4200518538/jobs/7286615050

```
6) [firefox] › page/page-click.spec.ts:162:3 › should click offscreen buttons ────────────────────

    Test timeout of 30000ms exceeded.

    page.click: Target closed
    =========================== logs ===========================
    waiting for locator('#btn5')
      locator resolved to <button id="btn5">5</button>
    attempting click action
      waiting for element to be visible, enabled and stable
      element is visible, enabled and stable
      scrolling into view if needed
      done scrolling
      performing click action
    ============================================================

      167 |     // We might've scrolled to click a button - reset to (0, 0).
      168 |     await page.evaluate(() => window.scrollTo(0, 0));
    > 169 |     await page.click(`#btn${i}`);
          |                ^
      170 |   }
      171 |   expect(messages).toEqual([
      172 |     'button #0 clicked',

        at /Users/runner/work/playwright/playwright/tests/page/page-click.spec.ts:169:16
    <inner error>
    Error: Target closed
    =========================== logs ===========================
    waiting for locator('#btn5')
      locator resolved to <button id="btn5">5</button>
    attempting click action
      waiting for element to be visible, enabled and stable
      element is visible, enabled and stable
      scrolling into view if needed
      done scrolling
      performing click action
    ============================================================
        at /Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/firefox/ffConnection.js:189:16
        at new Promise (<anonymous>)
        at FFSession.send (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/firefox/ffConnection.js:185:12)
        at RawMouseImpl.move (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/firefox/ffInput.js:91:24)
        at Mouse.move (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/input.js:153:23)
        at Mouse.click (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/input.js:197:26)
        at /Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/dom.js:468:109
        at /Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/dom.js:430:13
        at FrameManager.waitForSignalsCreatedBy (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/frames.js:114:26)
        at ElementHandle._performPointerAction (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/dom.js:424:57)
        at ElementHandle._retryPointerAction (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/dom.js:342:22)
        at /Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/frames.js:992:24
        at Frame.retryWithProgressAndTimeouts (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/frames.js:914:24)
        at /Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/frames.js:1012:29
        at ProgressController.run (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/progress.js:92:22)
        at FrameDispatcher.click (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/dispatchers/frameDispatcher.js:149:12)
        at DispatcherConnection.dispatch (/Users/runner/work/playwright/playwright/packages/playwright-core/lib/server/dispatchers/dispatcher.js:319:22)

    Pending operations:
      - page.click at tests/page/page-click.spec.ts:169:16
```

**Comment by mirao** [comment]
I have an issue with similar symptoms since I upgraded Playwright from 1.30 to 1.31 (and therefore I must downgrade back to 1.30, because I didn't find any workaround for it). Only Firefox is affected. Chromium and WebKit work fine.
A step in Firefox test randomly fails when clicking on an existing element (which is really clickable).

This is a debug output from CodeceptJS with Playwright:
```
04:23:51 [3.dev:firefox]     I click {xpath: .//*[contains(concat(' ', normalize-space(./@class), ' '), ' item ')][contains(., 'Courses')]}
04:30:10 [3.dev:firefox]  › <screenshotOnFail> Test failed, try to save a screenshot
04:30:10 [3.dev:firefox]  › Screenshot is saving to /home/node/workspace/tests/s4a/output/dev_firefox_3/Verify_you_91d4966d-120c-492c-977c-6f5022028d69.failed.png
04:30:10 [3.dev:firefox]   ✖ FAILED in 379797ms
...
04:55:23   1) LessonPathNoLessonPathDefined @noMobile
04:55:23        Verify you get no warning and a missing lesson path in the lesson:
04:55:23      elementHandle.click: Timeout 30000ms exceeded.
04:55:23 =========================== logs ===========================
04:55:23 attempting click action
04:55:23   waiting for element to be visible, enabled and stable
04:55:23   element is visible, enabled and stable
04:55:23   scrolling into view if needed
04:55:23   done scrolling
04:55:23   performing click action
04:55:23 ============================================================
04:55:23   elementHandle.click: Timeout 30000ms exceeded.
04:55:23   =========================== logs ===========================
04:55:23   attempting click action
04:55:23   
04:55:23   Scenario Steps:
04:55:23   - I.click({xpath: .//*[contains(concat(' ', normalize-space(./@class), ' '), ' item ')][contains(., 'Courses')]}) at Test.<anonymous> (./LessonPathNoLessonPathDefined_test.ts:225:11)
```

Unfortunately unable to reproduce reliably, but it never happens with Playwright 1.30.
Could it be the same issue as #20993?

**Comment by aslushnikov** [comment]
> Could it be the same issue as https://github.com/microsoft/playwright/issues/20993?

@mirao It might be. We'll see if yours goes away once this one is fixed.

**Comment by guillaumebrunerie** [comment]
I’m affected by this issue as well and needed to downgrade to Playwright 1.30. It only affects Firefox (and I'm on Mac OS, haven’t tried other OSs), the element does get clicked (it's a span that turns into a input when clicked and I see it turning into an input), but the `.click()` method seems to never resolve the promise.

**Comment by aslushnikov** [comment]
@guillaumebrunerie what version of MacOS do you run? Is it on M1? 

I was never able to reproduce this behavior on my mac. If you can share a repro that would repro on my local M1, that would help a ton!

**Comment by guillaumebrunerie** [comment]
@aslushnikov It's not an M1 unfortunately (MacBook Air 2019, MacOS Ventura 13.0.1). I can try to make a repro, but I have no easy access to a M1.

**Comment by guillaumebrunerie** [comment]
Here is a small repro of the issue I'm experiencing.

```js
import { test } from "@playwright/test";

test.only("should click", async ({page}) => {
	await page.evaluate(() => {
		const span = document.createElement("span");
		span.innerText = "abc";
		document.body.appendChild(span);
		span.addEventListener("pointerdown", (event) => {
			event.currentTarget.setPointerCapture(event.pointerId);
			document.body.removeChild(span);
		})
	});

	await page.getByText("abc").click();
});
```

It creates a span, and sets pointer capture when clicking on it while removing it from the DOM at the same time.
Works fine in Chromium and Webkit (in both 1.30 and 1.31), and works fine with Firefox in 1.30 but not in 1.31:
```

Running 3 tests using 2 workers
  1) [firefox] › bugs.spec.ts:3:6 › should click ───────────────────────────────────────────────────

    Test timeout of 30000ms exceeded.

    locator.click: Target closed
    =========================== logs ===========================
    waiting for getByText('abc')
      locator resolved to <span>abc</span>
    attempting click action
      waiting for element to be visible, enabled and stable
      element is visible, enabled and stable
      scrolling into view if needed
      done scrolling
      performing click action
    ============================================================

      13 | 	});
      14 |
    > 15 | 	await page.getByText("abc").click();
         | 	                            ^
      16 | });
      17 |

        at file:///.../bugs.spec.ts:15:30

    Pending operations:
      - locator.click at tests/playwright/bugs.spec.ts:15:30


  Slow test file: [firefox] › bugs.spec.ts (30.0s)
  Consider splitting slow test files to speed up parallel execution
  1 failed
    [firefox] › bugs.spec.ts:3:6 › should click ────────────────────────────────────────────────────
  2 passed (33.1s)
```

Maybe it's a different issue? But it also affects only Firefox on Playwright 1.31.

**Comment by faulpeltz** [comment]
> I have an issue with similar symptoms since I upgraded Playwright from 1.30 to 1.31 (and therefore I must downgrade back to 1.30, because I didn't find any workaround for it). Only Firefox is affected. Chromium and WebKit work fine. A step in Firefox test randomly fails when clicking on an existing element (which is really clickable).
> 
> Unfortunately unable to reproduce reliably, but it never happens with Playwright 1.30. Could it be the same issue as #20993?

@mirao Seeing this exact same behavior as well, Firefox only, fails to click a basic button, stuck in "Performing click action" then times out. Works with 1.30, stopped working with 1.31 (and still broken with 1.32)

**Comment by mirao** [comment]
@faulpeltz Thanks for info. So far I didn't test it with 1.32, but I'm going to try it out.
I can see this issue has a milestone `v1.33` so maybe it will be fixed in a next version.

**Comment by mirao** [comment]
> @faulpeltz Thanks for info. So far I didn't test it with 1.32, but I'm going to try it out.

My issue is present even in 1.32, just tested it.

**Comment by LorenzoBloedow** [comment]
This is also happening to me (v1.32.1), I have 3 clicks, two of them **don't** change the DOM structure of the button being clicked and they get executed just fine. The third one however, removes the button being clicked **after** it's clicked, on the `onPointerUp` event, so it should get executed just fine but instead it gets stuck, here's the debug output for the third click:
```
  pw:api attempting click action +5ms
  pw:api   waiting for element to be visible, enabled and stable +0ms
  pw:api   element is visible, enabled and stable +39ms
  pw:api   scrolling into view if needed +1ms
  pw:api   done scrolling +3ms
  pw:api   performing click action +33ms
```
At least in my case, this seems this is related to changing/removing the button's DOM structure even though it happens after the click.
And yes the test works on all the other browsers, and also used to work fine on Firefox before I updated Playwright.

**Comment by aslushnikov** [comment]
@LorenzoBloedow  please file a separate issue with a snippet to repro this!

**Comment by aslushnikov** [comment]
This might be fixed with https://github.com/microsoft/playwright/issues/22082

**Comment by aslushnikov** [comment]
@mirao this particular one was fixed in 1.33, so could you please file separately? We'd also need some repro so that we can act upon it.
